### ISSUE TITLE
when creating an MCMC model using an existing db the traces are not loaded to individual nodes.

### ISSUE THREAD

**Issue by isofer** [pymc-devs/pymc3#90]
when creating an MCMC model using an existing db the traces are loaded to the model but not to individual nodes.

```
import pymc as pm

#create a model
x = pm.Normal('x',0,1)
mc = pm.MCMC([x],db='pickle', dbname='testing.db')

#sample
mc.sample(200)

#print trace
print "printing the trace from x before closeing the db"
print x.trace()[:10]

#close the db
mc.db.close()

#now, let's load the db again
db = pm.database.pickle.load('testing.db')

#create a new model
x = pm.Normal('x',0,1)
mc2 = pm.MCMC([x], db=db)

#print trace from mcmc
print "printing the trace of x from mc2"
print mc2.trace('x')[:10]

#print trace
"printing the trace of x from x"
print x.trace()
```

This is the ouptut:

```
In [41]: execfile('/Users/imrisofer/PySchizo/bla2.py')
Sampling: 100% [000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000] Iterations: 200
printing the trace from x before closeing the db
[-1.30381504  0.54872208  0.17317576  0.88309938 -0.3897846  -0.51572362
 -0.706447    0.10687589 -2.07283465  1.18359088]
printing the trace of x from mc2
[-1.30381504  0.54872208  0.17317576  0.88309938 -0.3897846  -0.51572362
 -0.706447    0.10687589 -2.07283465  1.18359088]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/Users/imrisofer/Documents/Projects/rapid/alz/<ipython-input-41-3b0704b3bada> in <module>()
----> 1 execfile('/Users/imrisofer/PySchizo/bla2.py')

/Users/imrisofer/PySchizo/bla2.py in <module>()
     30 #print trace

     31 "printing the trace of x from x"
---> 32 print x.trace()
     33 
     34 

TypeError: 'bool' object is not callable
```

this kind of bug creates problems when trying to plot a single node or when using any other function that's try to excess the node's trace.

**Comment by fonnesbeck** [comment]
It does get assigned to the node, but not until sampling is executed:

```
In [3]: mc2.sample(10)
Sampling: 100% [00000000000000000000000000000000000000000000000] Iterations: 10

In [4]: x.trace()
Out[4]: 
array([ 0.39246356,  2.15682678, -1.38975955, -0.83109021, -0.38935031,
    0.65201667,  1.32492933, -1.87418302,  1.65249373,  1.6733377])
```

Can you work with this, or is it critical that they be available before sampling?

**Comment by isofer** [comment]
I believe it is critical to have access before sampling for a couple of
reasons:
1) when you sample again you create another chain. the problem is that some
function works only on the last chain (for instance, pm.Matplotlib.plot).
so these functions won't work on the chain that was saved in the db.
2) I work on a module that uses pymc. some of the functions and attributes
I wrote losse their functionality when the model is loaded from a db,
because of this problem.

thanks

On Tue, Jan 31, 2012 at 5:11 PM, Chris Fonnesbeck <
reply@reply.github.com

**Comment by fonnesbeck** [comment]
OK, I pushed at least a temporary fix in 1d2657a0f13d106a6268441b2695815784d3680e

Please test.

**Comment by isofer** [comment]
seems to work great. thanks.

On Tue, Jan 31, 2012 at 9:21 PM, Chris Fonnesbeck <
reply@reply.github.com
